Mastodon/Scene/Onboarding/Login/MastodonLoginViewController.swift

```diff
@@ -14,279 +14,305 @@ import AuthenticationServices
-
-      viewModel.selectedServer = .init(domain: domain, instance: .init(domain: domain))
-      contentView.navigationActionView.nextButton.isEnabled = true
-    } else {
-      viewModel.selectedServer = nil
-      contentView.navigationActionView.nextButton.isEnabled = false
     }
-  }
-
-  // MARK: - Notifications
-  @objc func keyboardWillShowNotification(_ notification: Notification) {
-
-    guard let userInfo = notification.userInfo,
-          let keyboardFrameValue = userInfo[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue,
-          let duration = userInfo[UIResponder.keyboardAnimationDurationUserInfoKey] as? NSNumber
-    else { return }
-
-    // inspired by https://stackoverflow.com/a/30245044
-    let keyboardFrame = keyboardFrameValue.cgRectValue
-
-    let keyboardOrigin = view.convert(keyboardFrame.origin, from: nil)
-    let intersectionY = CGRectGetMaxY(view.frame) - keyboardOrigin.y;
-
-    if intersectionY >= 0 {
-      contentView.bottomConstraint?.constant = intersectionY - view.safeAreaInsets.bottom
+    
+    // MARK: - Notifications
+    @objc func keyboardWillShowNotification(_ notification: Notification) {
+        
+        guard let userInfo = notification.userInfo,
+              let keyboardFrameValue = userInfo[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue,
+              let duration = userInfo[UIResponder.keyboardAnimationDurationUserInfoKey] as? NSNumber
+        else { return }
+        
+        // inspired by https://stackoverflow.com/a/30245044
+        let keyboardFrame = keyboardFrameValue.cgRectValue
+        
+        let keyboardOrigin = view.convert(keyboardFrame.origin, from: nil)
+        let intersectionY = CGRectGetMaxY(view.frame) - keyboardOrigin.y;
+        
+        UIView.animate(withDuration: duration.doubleValue, delay: 0, options: .curveEaseInOut) {
+            self.view.layoutIfNeeded()
+        }
     }
-
-    UIView.animate(withDuration: duration.doubleValue, delay: 0, options: .curveEaseInOut) {
-      self.view.layoutIfNeeded()
+    
+    @objc func keyboardWillHideNotification(_ notification: Notification) {
+        
+        guard let userInfo = notification.userInfo,
+              let duration = userInfo[UIResponder.keyboardAnimationDurationUserInfoKey] as? NSNumber
+        else { return }
+        
+        UIView.animate(withDuration: duration.doubleValue, delay: 0, options: .curveEaseInOut) {
+            self.view.layoutIfNeeded()
+        }
     }
-  }
-
-  @objc func keyboardWillHideNotification(_ notification: Notification) {
-
-    guard let userInfo = notification.userInfo,
-          let duration = userInfo[UIResponder.keyboardAnimationDurationUserInfoKey] as? NSNumber
-    else { return }
-
-    contentView.bottomConstraint?.constant = 0
+}
 
-    UIView.animate(withDuration: duration.doubleValue, delay: 0, options: .curveEaseInOut) {
-      self.view.layoutIfNeeded()
+private extension MastodonLoginViewController {
```

Comment:
No need to change, but a question: What's the reason to put `setRightBarButtonState(_:)` into its own extension?

Reply:
Tbh, no real reason, I'll move it to the class directly. I actually like to split things e.g. all things private. But given this is the only function I don't see a strong reason to do this.